### DISCUSSION TITLE
Combined glow-tts and universal hifigan result in noise

### DISCUSSION THREAD

**Discussion by Wayne-wonderai** [coqui-ai/TTS#1012] [General Q&A]
Hello, I'm trying to use glow-tts and [Universal HifiGAN](https://drive.google.com/drive/folders/1_TmA2ramp42jhqGhFEn_Vivyo65uH8MV), but the generated voice is just noise, can you help me find out the problem?
Here's my code.(Just a copy and a little change from [synthesize.py](https://github.com/coqui-ai/TTS/blob/main/TTS/bin/synthesize.py)

```
import os
import argparse
import sys
from argparse import RawTextHelpFormatter
from pathlib import Path
os.environ["CUDA_VISIBLE_DEVICES"]="0"
from TTS.utils.manage import ModelManager
from TTS.utils.synthesizer import Synthesizer
sentence = "Here is your daily weather report. Cloudy with a slight chance of rain so don't forget to take an umbrella with you!"
filename = 'test_1'
path = '../.models.json'
manager = ModelManager(path)

model_name = 'tts_models/en/ljspeech/glow-tts'
config_path = None
speakers_file_path = None
vocoder_path = None
vocoder_config_path = None
encoder_path = None
encoder_config_path = None

model_path, config_path, model_item = manager.download_model(model_name)
vocoder_path = 'model/hifigan/LibriTTS-360-100-clean/best_model_458080.pth.tar'
vocoder_config_path = 'model/hifigan/LibriTTS-360-100-clean/config.json'

# load models
synthesizer = Synthesizer(
    model_path, 
    config_path, 
    speakers_file_path,
    vocoder_path,
    vocoder_config_path,
    encoder_path,
    encoder_config_path,
    False,
)

speaker_idx = None
speaker_wav = None
print(" > Text: {}".format(sentence))
wav = synthesizer.tts(sentence , speaker_idx, speaker_wav)
save_path = 'save_wav'

print(" > Saving output to {}".format(save_path+'/'+filename))
synthesizer.save_wav(wav, 'save_wav/'+filename+'.wav')
```
And the result as attach.
Thanks!
[test_1.zip](https://github.com/coqui-ai/TTS/files/7702588/test_1.zip)

**Comment by gullabi** [comment]
Hi, I had the same problem initially. Different text-spec models have different audio parameters, same in vocoders. Hence glow-tts is not compatible with hifigan_v2. You can check the audio values in the configs of both and see the differences. If I am not mistaken, glow-tts should work with multi-melgan and univnet.

**Reply by Wayne-wonderai** [reply]
In this [Demo](https://jaywalnut310.github.io/glow-tts-demo/index.html), it combined glow-tts with hifiganv2. What I'm trying is to reproduce these results. (as I thought hifigan may generate better sound quality)
And yes, glow-tts work with multi-melgan and univnet well.

**Reply by gullabi** [reply]
You should compare the audio parameters in the configs of your vocoder and your text-spec model. There are mismatches probably in the min max mel frequencies
![config](https://user-images.githubusercontent.com/40303490/145988214-00c687dd-a786-4e9a-b331-24a1d8c83d83.png)
.

**Reply by Wayne-wonderai** [reply]
ok, I'll open another discussion for how to choose the most suitable mel_fmax & mel_fmin for our customized dataset
